src/bin/nydus-image/decompress.rs

```diff
@@ -0,0 +1,1158 @@
+extern crate tar;
+
+use std::{
+    collections::HashMap,
+    ffi::OsStr,
+    fs::{File, OpenOptions},
+    io::{self, Cursor, Error, ErrorKind, Read},
+    iter::{self, from_fn, repeat},
+    os::unix::prelude::{OsStrExt, OsStringExt},
+    path::{Path, PathBuf},
+    rc::Rc,
+    str,
+    sync::Arc,
+    vec::IntoIter,
+};
+
+use anyhow::Result;
+use nydus_api::http::LocalFsConfig;
+use nydus_rafs::{
+    metadata::{RafsInode, RafsMode, RafsSuper},
+    RafsIoReader,
+};
+use nydus_utils::compress::{self, Algorithm};
+use storage::{
+    backend::{localfs::LocalFs, BlobBackend, BlobReader},
+    device::{BlobChunkInfo, BlobInfo},
+    utils::alloc_buf,
+};
+use tar::{Builder, EntryType, Header};
+
+use crate::core::node::InodeWrapper;
+
+static PAX_SEP1: &[u8; 1] = b" ";
+static PAX_SEP2: &[u8; 1] = b"=";
+static PAX_PREFIX: &[u8; 13] = b"SCHILY.xattr.";
+static PAX_DELIMITER: &[u8; 1] = b"\n";
+
+pub trait Decompressor {
```

Comment:
> Seems we are implementing pack/unpack(tar/untar) instead of compress/decompress, should we name it as Unpacker or something similar?

Unpack is more comfortable~. Snapshotter will use `unpack` instead of `decompress`.